src/main/java/org/folio/service/file/S3JobRunningVerticle.java

```diff
@@ -0,0 +1,329 @@
+          new StatusDto().withStatus(StatusDto.Status.PROCESSING_IN_PROGRESS),
+          params
+        )
+          .map(v -> job)
+      )
+      .compose(this::downloadFromS3)
+      .compose(job ->
+        fileProcessor
+          .processFile(
+            job.getFile(),
+            job.getJobExecution().getId(),
+            // this is the only part used on our end
+            new JobProfileInfo()
+              .withDataType(
+                JobProfileInfo.DataType.fromValue(
+                  job.getQueueItem().getDataType()
+                )
+              ),
+            params
+          )
+          .map(v -> job)
+      )
+      .onFailure((Throwable err) -> {
+        LOGGER.error("Unable to start chunk {}", queueItem, err);
+        err.printStackTrace();
+
+        updateJobExecutionStatusSafely(
+          queueItem.getJobExecutionId(),
+          new StatusDto()
+            .withErrorStatus(ErrorStatus.FILE_PROCESSING_ERROR)
+            .withStatus(StatusDto.Status.ERROR),
+          params
+        );
+      })
+      .onSuccess((QueueJob result) ->
+        LOGGER.info(
+          "Completed processing job execution {}!",
+          queueItem.getJobExecutionId()
+        )
+      )
+      .onComplete(v -> {
+        queueItemDao.deleteDataImportQueueItem(queueItem.getId());
+
+        File file = localFile.get();
+        if (file != null) {
+          try {
+            FileUtils.delete(file);
+          } catch (IOException e) {
+            LOGGER.error(
+              "Could not clean up temporary file {}: ",
+              file.toPath(),
+              e
+            );
+          }
+        }
+      });
+  }
+
+  protected Future<Void> updateJobExecutionStatusSafely(
+    String jobExecutionId,
+    StatusDto status,
+    OkapiConnectionParams params
+  ) {
+    return uploadDefinitionService
+      .updateJobExecutionStatus(jobExecutionId, status, params)
+      .map((Boolean successful) -> {
+        if (Boolean.FALSE.equals(successful)) {
+          LOGGER.error(
+            "Unable to change job {} status to {}",
+            jobExecutionId,
+            status
+          );
+          throw new IllegalStateException(
+            "Unable to update job execution status"
+          );
+        }
+        return successful;
+      })
+      .mapEmpty();
+  }
+
+  protected Future<QueueJob> downloadFromS3(QueueJob job) {
+    return minioStorageService
+      .readFile(job.getJobExecution().getSourcePath())
+      .map((InputStream inputStream) -> {
+        try (
+          InputStream autoCloseMe = inputStream;
+          OutputStream outputStream = new FileOutputStream(job.getFile())
+        ) {
+          inputStream.transferTo(outputStream);
+
+          return job;
+        } catch (IOException e) {
+          throw new UncheckedIOException(e);
+        }
+      });
+  }
+
+  protected File createLocalFile(DataImportQueueItem queueItem) {
+    try {
+      File localFile = Files
```

Comment:
Use vertx.fileSystem() to create files or run this on a worker thread to not block the event loop.